### ISSUE TITLE
question: use msc_rules_cleanup not free memory

### ISSUE THREAD

**Issue by Brhetty** [SpiderLabs/ModSecurity#2636]
hi:
    I am just add rule but use msc_rule_cleanup bot free memory. What's wrong with me

modsecurity vserion: v3.0.4
```
int main()
{
    const char *error;
    const char *rule = "SecRuleEngine On";
    void *rules_set;
    int i = 0;
    int rc;
    rules_set = msc_create_rules_set();
    for(i = 0; i < 100000; i++) {
       rc =  msc_rules_add(rules_set, rule, &error);
       if (rc < 0) {
           printf("error: %s", error);
       }
    }
    msc_rules_cleanup(rules_set);
}
```

**Comment by martinhsv** [comment]
Hello @Brhetty ,

I do not understand what exactly you are asking.  Would you like to try rephrasing or expanding on your question?

**Comment by Brhetty** [comment]
> Hello @Brhetty ,
> 
> I do not understand what exactly you are asking. Would you like to try rephrasing or expanding on your question?

When I use msc_rules_add, it will  memory leak.Could  you help me?

**Comment by martinhsv** [comment]
Since you're using a older version of ModSecurity, a normal piece of advice would be to upgrade; there were at least some memory problems fixed between v3.0.4 and v3.0.5.

However, in this case, I assume you are simply triggering previously-reported issues around memory leaks when nginx rule reload is performed without a restart. See for example #2552
